services/PostService.js

```diff
@@ -26,6 +26,21 @@ module.exports = class PostService {
     }
   }
 
+  static async searchPostWithId(id) {
+    try {
+      const post = await Post.findAll({
+        where: {
+          id: {
```

Comment:
this is findAll not findOne, https://sequelize.org/docs/v6/core-concepts/model-querying-finders/